bin/cron/update_census_mapbox

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env ruby
+
+# The map on code.org/yourschool is run off of Mapbox
+# This script updates the data on Mapbox by querying all rows from
+# census_summaries table for the appropriate school year,
+# creating maptiles with Mapbox's command line tool, Tippecanoe,
+# and uploading them to Mapbox's server.
+
+# After setting up authentication, provide the following CDO.* attributes:
+#   CDO.mapbox_upload_token: (the auth token for uploading tiles to Mapbox's servers)
+
+require_relative '../../dashboard/config/environment'
+require 'cdo/properties'
+
+require_relative 'upload_to_mapbox'
+
+def make_location(lat, long)
+  return nil if lat.nil? || long.nil?
+
+  "#{format('%.6f', lat)}, #{format('%.6f', long)}"
+end
+
+# Update summaries before updating the fusion table
+# If any single teaches_cs category in a given year changes by more than
+# 10% of the total number of schools then issue a Honeybadger notification.
+threshold = School.count * 0.1
+old_summary_counts = Census::CensusSummary.group(:school_year, :teaches_cs).count
+Census::CensusSummary.summarize_census_data
+new_summary_counts = Census::CensusSummary.group(:school_year, :teaches_cs).count
+
+unless old_summary_counts.select {|k, c| (c - new_summary_counts[k]).abs > threshold}.empty?
+  Honeybadger.notify(
+    error_message: "High rate of change in census_summaries. This may be expected if summary logic was recently changed or if a new data source was just added.",
+    error_class: "Census Summary High Rate of Change",
+    context: {
+      old_counts: old_summary_counts,
+      new_counts: new_summary_counts
+    }
+  )
+end
+
+# The map should show the data from the school year that began
+# in the previous calendar year. So, Aug-Dec we show the previous school
+# year's data and then switch over to the current school year in Jan.
+year = Date.today.year - 1
+geojson = JSON.parse('{ "type": "FeatureCollection", "features": [] }')
+Census::CensusSummary.where(school_year: year).find_each do |summary|
+  school = summary.school
+  geojson["features"] <<
+    {
+      geometry: {
+        coordinates: [school.longitude.to_f, school.latitude.to_f],
+        type: "Point"
+      },
+      properties: {
+        school_id: school.id,
+        year: year,
+        school_name: school.name.titleize,
+        school_city: school.city.titleize,
+        school_state: school.state,
+        teaches_cs: summary.teaches_cs
+      },
+      type: "Feature"
+    }
+end
+
+file = Tempfile.new(['census', 'geojson'])
+tiles = Tempfile.new(['censustiles', 'mbtiles'])
+file.write(JSON.pretty_generate(geojson))
+stdout, _status = Open3.capture2(
+  "tippecanoe -Z 1 -rg --cluster-distance=2 -o #{tiles.path} -l \"census\" --force #{file.path}"
+)
+upload_maptiles(tiles.path, 'censustiles')
+puts stdout
```

Comment:
I think that standard output from a cron job is discarded these days (as we discovered in https://github.com/code-dot-org/code-dot-org/blob/staging/bin/cron/deliver_poste_messages)... should we have some honeybadger (or other) error reporting if anything goes wrong in this process?

Reply:
I think best practice in a cron job is to use `CDO.log`.  I think if anything is written to standard out by a cron job (such as `puts`), our wrapper that invokes cron scripts will notify HoneyBadger.

Reply:
Maybe we should add a lint recommending `CDO.log` for any invocation of `puts` in a cron job?

Reply:
That would be great if true, and if we can verify.  In the above-mentioned file we actually switched from `CDO.log` to honeybadger explicitly (in https://github.com/code-dot-org/code-dot-org/pull/29210) and I believe it was because we weren't seeing those log statements being recorded anywhere.

Reply:
I expanded on this in both files to notify with Honeybadger if any of the steps fail.

Reply:
It's possible that the `cloudwatch-logger` service on `production-daemon`, which streams logs to CloudWatch stopped working.  It seems to have failed on 6/11/2019 (that's the [last event](https://console.aws.amazon.com/cloudwatch/home?region=us-east-1#logStream:group=production-syslog;prefix=production-daemon;streamFilter=typeLogStreamPrefix) logged).  That would explain why you weren't seeing those logs anywhere.

[Here's an example ](https://console.aws.amazon.com/cloudwatch/home?region=us-east-1#logEventViewer:group=production-syslog;stream=production-daemon-19339;filter=contact;start=2019-06-10T20:28:48Z) of `CDO.log` correctly posting to CloudWatch from a cron job.

I've restarted the logger service on `production-daemon`.  I'll bump up the priority of the [Jira Task](https://codedotorg.atlassian.net/browse/INF-151) to improve the reliability of this service.